src/node/stage/approved.rs

```diff
@@ -902,18 +1016,95 @@ impl Approved {
             return true;
         }
 
+        if new_infos.len() > 1 {
+            debug!("splitting with new_infos {:?}", new_infos);
+        }
+
         for info in new_infos {
             let participants: BTreeSet<_> = info.elder_ids().copied().collect();
-            let _ = self.dkg_cache.insert(participants.clone(), info);
-            self.consensus_engine.vote_for(NetworkEvent {
-                payload: AccumulatingEvent::StartDkg(participants),
-                proof_share: None,
-            });
+            let parsec_version = self.consensus_engine.parsec_version();
+            let dkg_key = (participants.clone(), parsec_version);
+
+            if let Some(dkg_result) = self.dkg_voter.push_info(&dkg_key, info) {
+                // Got notified of the sibling DKG result, happens during split.
+                if let Err(err) = self.handle_dkg_result_event(core, &participants, &dkg_result) {
+                    debug!(
+                        "Failed handle sibling notified dkg_result {:?} - {:?}",
+                        dkg_key, err
+                    );
+                    self.dkg_voter.insert_dkg_result(dkg_key, dkg_result);
+                }
+                continue;
+            }
+
+            // In case all the current elders split into one side of the sub-section, triggering the
+            // sibling section's DKG process by send them an Initial DKG message.
```

Comment:
I wonder whether we should do this always, not just when all elders went only to one subsection. For example, say 6 elders went to one subsection and 1 to the other, but that 1 is faulty/malicious and never broadcasts the necessary dkg messages to the rest. Wouldn't that be a problem?

Reply:
Yes I think we should always do this and then we are consistent.

Reply:
yes, it could be a problem.
Was just want to reduce some messaging.
will make the notification happens always.

Reply:
Later I would like to force new node names when nodes become elders. So we try and spread them evenly in the address space. IT will help but not guarantee this never happens.